index.html

```diff
@@ -0,0 +1,293 @@
+    </p>
+    <pre class="idl">
+    dictionary IdentityRequestOptions {
+      AbortSignal signal;
+      required sequence&lt;IdentityRequestProvider&gt; providers;
+    };
+    </pre>
+    <p>
+      The <dfn data-dfn-for="IdentityRequestOptions">providers</dfn> member is
+      a sequence of [=digital credential/request protocol=] specific providers.
+    </p>
+    <h3>
+      The `signal` member
+    </h3>
+    <p>
+      The <dfn data-dfn-for="IdentityRequestOptions">signal</dfn> member is an
+      {{AbortSignal}} that allows the caller to abort an ongoing [=request
+      digital identity credentials=].
+    </p>
+    <h2>
+      The `IdentityRequestProvider` dictionary
+    </h2>
+    <p>
+      The {{IdentityRequestProvider}} dictionary is used to specify a
+      [=digital credential/request protocol=] specific provider, while also handling
+      cryptographic requirements.
+    </p>
+    <pre class="idl">
+    dictionary IdentityRequestProvider {
+      required DOMString protocol;
+      required ArrayBuffer request;
+      required CryptoKey publicKey;
+    };
+    </pre>
+    <h3>
+      The `protocol` member
+    </h3>
+    <p>
+      The <dfn data-dfn-for="IdentityRequestProvider">protocol</dfn> member
+      denotes the [=digital credential/request protocol=] when requesting an
+      identify credential via {{CredentialsContainer/requestIdentity()}}.
+    </p>
+    <p>
+      The {{IdentityRequestProvider/protocol}} member's value is be one of the
+      keys defined in [[[#protocol-registry]]].
+    </p>
+    <h3>
+      The `request` member
+    </h3>
+    <p>
+      The <dfn data-dfn-for="IdentityRequestProvider">request</dfn> member is
+      the request to be sent to the [=identity credential provider=].
+    </p>
+    <h3>
+      The `publicKey` member
+    </h3>
+    <p>
+      The <dfn data-dfn-for="IdentityRequestProvider">publicKey</dfn> member is
+      the public key of the [=identity credential provider=].
+    </p>
+    <h2>
+      Requesting a digital identity credential
+    </h2>
+    <p>
+      The following steps are taken to <dfn>request digital identity
+      credentials</dfn>, with {{IdentityRequestOptions}} |options|:
+    </p>
+    <ol>
+      <li>Let |promise| be [=a new promise=].
+      </li>
+      <li>Return |promise| and continue running the following steps [=in
+      parallel=].
+      </li>
+      <li>For each |provider:IdentityRequestProvider| of |options|'s
+      {{IdentityRequestOptions/providers}}.
+        <ol>
+          <li>[=verify the public key=] of |provider|.
+          </li>
+        </ol>
+      </li>
+      <li>...present [=identity credential provider=] to user.
+      </li>
+    </ol>
+    <h2>
+      The `DigitalCredential` interface
+    </h2>
+    <p>
+      The {{DigitalCredential}} interface represents a [=digital identity
+      credential=].
+    </p>
+    <pre class="idl">
+    [Exposed=Window, SecureContext]
+    interface DigitalCredential : Credential {
```

Comment:
I think inheriting from `Credential` would create a problem for implementations, because, according to the Credential Manager spec, implementations MUST loop through the subclasses and look the internal methods.

https://www.w3.org/TR/credential-management-1/#credential-internal-methods

```
Unless otherwise specified, each [interface object](https://heycam.github.io/webidl/#dfn-interface-object) created for interfaces which [inherit](https://heycam.github.io/webidl/#dfn-inherit) from [Credential](https://www.w3.org/TR/credential-management-1/#credential) MUST provide implementations for at least one of these internal methods, overriding [Credential](https://www.w3.org/TR/credential-management-1/#credential)'s default implementations, as appropriate for the [credential](https://www.w3.org/TR/credential-management-1/#concept-credential) type. E.g., [§3.2 The PasswordCredential Interface](https://www.w3.org/TR/credential-management-1/#passwordcredential-interface), [§4.1 The FederatedCredential Interface](https://www.w3.org/TR/credential-management-1/#federatedcredential-interface), and [[WEBAUTHN]](https://www.w3.org/TR/credential-management-1/#biblio-webauthn).
```

We could emphasize on the "Unless otherwise specified" part, but the challenge here is that this `DigitalCredential` here really isn't a Credential Manager `Credential` in the sense that all of the other ones are (PublicKeyCredential, PasswordCredential, FederatedCredential and IdentityCredential).

So, I think we need to make a decision as to whether we want to integrate properly with the Credential Management API (e.g. implement the other methods, like `[[CollectFromCredentialStore]]`) or not (e.g. not subclass `Credential` here).

WDYT?

Reply:
If these APIs end up very different, and the only difference in their names is the word `Digital` I think that will lead to a lot of bug reports and developer confusion.

Reply:
How much of the other methods of CredMan API do we think digital credentials API will reuse?
It has been pointed out number of times that "credential" in a conventional/CredMan refers to the things like password, wubauthn keypair, which is different from how "verifiable/digital credentials" refer to collection of attributes of the user attested by the issuer. It might be that despite all this, a lot of the same methods can be reused, I don't know, but that sounds like something that could influence the decision here.

Reply:
@samuelgoto, I agree that we should be building on the Credential Manager spec. The goal is be to eventually copy/paste this over to that spec. And yes, I agree that integration needs to be coherent and we are not doing a separate thing. 

At the same time, there was good reason for not using `.get()`: we can more strictly restrict the call site to just `DigitalCredentialRequestOptions` without allowing the mixing of credential types in the request (yes, we could also enforce that in `.get()`, but it's a bit more messy because it can't be enforced with WebIDL.... nevertheless, whatever we do spec should be `.get()` compatible). Also, calling `.requestIdentity()` was supposed to make it clear to developers which sheet/UI is going to be show: the one to pick a driver's license or similar credential, not for passwords or whatever other things are possible with straight up `.get()`. 

About who we should talk to, I think we should just talk to Credential Manager API folks... speaking as a ex-TAG member, I don't think the TAG will have much to offer here TBH (this is outside their purview, and can only provide very surface guidance... and will probably just tell us to go to talk to WebAppSec 🫠).

Reply:
Chiming in here that making this extend `Credential` would be something I believe we explicitly looked to avoid with the decision to use `.requestIdentity()`. If we subclass, that makes the more "login" centric `.get()` able to request an identity.

I think that the options we have are (1) making a parallel construction for driver's-license like things where we reuse algorithms as helpers or (2) making the `Credential` inherit a more generic object that we lift the shared bits into.

Reply:
>  I think we should just talk to Credential Manager API folks

Yeah, I think that's the right next step.

> that makes the more "login" centric .get()

I think there are a couple of things that I'm struggling with this proposition:

1) First, I don't share the interpretation that the Credential Manager API is for "login".
2) Second, it is clear to me that the Digital Credentials here are going to be used for "login": it even shows up in regulation.

I think some of my confusion here is that i think maybe we don't quite understand what we mean by "login" (e.g. i think there is a big distinction between sign-up - i.e. account creation and recovery -, and sign-in).

Chrome expects FedCM to be used for a lot of [use cases](https://github.com/fedidcg/FedCM/blob/main/meetings/2021/Web%20Identity%20API.pdf) that I don't think I would call "login", so that's what I'm struggling to reconcile.

Reply:
By login I mean sign-in.  

> First, I don't share the interpretation that the Credential Manager API is for "login".

Sign-in is mentioned in the first sentence of the use cases section of that spec and the opening sentence of that spec is "Signing into websites is more difficult than it should be." 

> Second, it is clear to me that the Digital Credentials here are going to be used for "login": it even shows up in regulation.

I'm not a lawyer so I can't speak to what the EU is mandating, but is it something that we should encourage globally? I say not.

> Chrome expects FedCM to be used for a lot of [use cases](https://github.com/fedidcg/FedCM/blob/main/meetings/2021/Web%20Identity%20API.pdf) that I don't think I would call "login", so that's what I'm struggling to reconcile.

IMO the non-sign-in use cases covered by FedCM can be explained as the consequences of trying to sign in while simultaneously dealing with an out-of-the-page IDP in the best way possible.

Reply:
Ok, I'm going to resolve this, since @bvandersloot-mozilla 's [alternative design](https://github.com/WICG/digital-credentials/issues/61#issuecomment-1921637318) really captures the essence of what I was struggling with, which is now reflected in this PR.